common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -157,6 +159,98 @@ class @Problem
 
   # TODO add hooks for problem types here by inspecting response.html and doing
   # stuff if a div w a class is found
+          if response.progress_status != "done"
+            @maybeSetupTimer()
+
+  getSecondsLeft: =>
+    # Returns 
+    #   a nonnegative integer representing seconds left on a timed exam.
+    #   -1 on error
+    if _.isUndefined(@startTime) or _.isUndefined(@totalSeconds)
+      return -1
+    else
+      return Math.floor(Math.max(@totalSeconds - ((new Date() - @startTime) / 1000), 0))
+
+  maybeSetupTimer: =>
+    if @el.find(".problem-timer").length
+      @$timer = @el.find('.problem-timer')
+      @$display = @el.find('.minutes-left')
+      @startTime = new Date()
+      @totalSeconds = parseInt(@$timer.data('totalSecondsLeft'), 10)
+      @secondsBeforeWarning = parseInt(@$timer.data('secondsBeforeWarning'), 10)
+      @submittedBeforeTimeExpired = false
+
+      # The timer is initially hidden by CSS 
+      @$timer.show()
+
+      # Clear old timers, eg. if we press the submit button
+      if @timerId
+        clearInterval(@timerId)
+
+      # Sync every second
+      @timerId = setInterval(@syncTimer, 1000)
+
+      # Initialize and show timer
+      @syncTimer()
```

Comment:
Yes, I understand.  My consider is really that `data-total-seconds-left` is set in problem.html, and I'm wondering whether this is statically on the page (i.e. the browser will just use the cached page and value if the user hits the back button) or "dynamic", i.e. problem.html is re-fetched from the server via ajax.  I have a feeling if this is static, the browser will use the cached value, the time remaining display will be wrong.

Have you (manually) tested this case?